noir-projects/noir-contracts/contracts/router_contract/src/main.nr

```diff
@@ -0,0 +1,41 @@
+/// The purpose of this contract is to perform a check in public without revealing what contract enqued the public
+/// call. This is achieved by having a private function on this contract that enques the public call and hence
+/// the `msg_sender` in the public call is the address of this contract.
+contract Router {
+    use aztec::utils::comparison::assert_comparison;
+
+    // docs:start:check_timestamp
+    /// Enqueues a public call that asserts that `lhs` (left side of the comparison) timestamp satisfies
+    /// the `operation` with respect to the current timestamp.
+    #[aztec(private)]
```

Comment:
could we skip an interation of the kernel by the main contract directly calling `_check_timestamp_public()`

Reply:
it obviously leaks the original contract thoughh hmmph..